api/filters.go

```diff
@@ -356,98 +420,106 @@ func (api *FilterAPI) addFilterBlueprintDimensionOption(w http.ResponseWriter, r
 	}
 
 	if err := api.dataStore.AddFilterDimensionOption(addDimensionOption); err != nil {
-		log.Error(err, log.Data{"dimension_option": addDimensionOption})
+		log.Error(err, logData)
 		setErrorCode(w, err)
 		return
 	}
 
 	setJSONContentType(w)
 	w.WriteHeader(http.StatusCreated)
 
-	log.Info("created new dimension option for filter blueprint", log.Data{"dimension_option": addDimensionOption})
+	log.Info("created new dimension option for filter blueprint", logData)
 }
 
 func (api *FilterAPI) updateFilterBlueprint(w http.ResponseWriter, r *http.Request) {
 	vars := mux.Vars(r)
 	filterID := vars["filter_blueprint_id"]
-	var submitted string
-	submitted = r.URL.Query().Get("submitted")
+	submitted := r.URL.Query().Get("submitted")
+	logData := log.Data{"func": "updateFilterBlueprint", "filter_blueprint_id": filterID, "submitted": submitted}
+	log.Info("updating filter blueprint", logData)
 
+	const FilterSubmitted = "true"
 	filter, err := models.CreateFilter(r.Body)
 	if err != nil {
 
 		// When filter blueprint has query parameter `submitted` set to true then
 		// request can have an empty json in body for this PUT request
-		if submitted != "true" || err != models.ErrorNoData {
-			log.Error(err, log.Data{"filter_blueprint_id": filterID})
+		if submitted != FilterSubmitted || err != models.ErrorNoData {
+			log.Error(err, logData)
 			http.Error(w, badRequest, http.StatusBadRequest)
 			return
 		}
 	}
 
 	filter.FilterID = filterID
+	logData["filter_update"] = filter
+
+	if err = models.ValidateFilterBlueprintUpdate(filter); err != nil {
+		log.Error(err, logData)
+		http.Error(w, badRequest, http.StatusBadRequest)
+		return
+	}
 
 	currentFilter, err := api.dataStore.GetFilter(filterID)
 	if err != nil {
-		log.Error(err, log.Data{"filter_blueprint": filter, "filter_blueprint_id": filterID})
+		log.Error(err, logData)
 		setErrorCode(w, err)
 		return
 	}
+	logData["current_filter"] = currentFilter
 
-	newFilter := currentFilter
-	if filter.InstanceID != "" {
+	newFilter, VersionHasChanged := createNewFilter(filter, currentFilter)
```

Comment:
Should this variable really be exported?

Reply:
Nope, just a typo, I will remove it